### PR TITLE
Add --format option to buildah task

### LINKED ISSUE #221: Support buildah --format=docker option for non OCI compliant registries
# Expected Behavior
Buildah `Task` should support the [buildah --format option](https://github.com/containers/buildah/blob/master/docs/tutorials/03-on-build.md#using-onbuild-in-buildah) to be able to set it to 'docker' in order to push to registries that are not supporting OCI yet (e.g. harbor, see https://github.com/goharbor/harbor/issues/8268).

Add a `FORMAT` param to the buildah `Task` so users can specify the format and use 'docker'.

```yaml
params:
  - name: FORMAT
     value: "docker"
```
# Actual Behavior
Images are alway with built with format oci, which is the default.

# Steps to Reproduce the Problem

1. Build an image with buildah task
2. Try Push it to a registry without OCI support

# Additional Info
I will provide a PR with the option available.

### PR DESCRIPTION
**Pull request by joachimprinzbach**
# Changes

Add --format option to buildah task, closes #221 

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)
- [x] Yaml file complies with [yamllint](https://github.com/adrienverge/yamllint) rules.

_See [the contribution guide](https://github.com/tektoncd/catalog/blob/master/CONTRIBUTING.md)
for more details._

### CODE REVIEW

File: buildah/README.md [hunk 1]
```diff
@@ -25,6 +25,8 @@ kubectl apply -f https://raw.githubusercontent.com/tektoncd/catalog/master/build
   `.`)
 * **TLSVERIFY**: Verify the TLS on the registry endpoint (for push/pull to a
   non-TLS registry) (_default:_ `true`)
+* **FORMAT**: the format of the built container, oci or docker (_default:_
```
Comment (navidshaikh) [inline buildah/README.md]:
```suggestion
* **FORMAT**: The format of the built container, oci or docker (_default:_
```

File: buildah/buildah.yaml [hunk 2]
```diff
@@ -18,6 +18,9 @@ spec:
     - name: TLSVERIFY
       description: Verify the TLS on the registry endpoint (for push/pull to a non-TLS registry)
       default: "true"
+    - name: FORMAT
+      description: the format of the built container, oci or docker
```
Comment (navidshaikh) [inline buildah/buildah.yaml]:
```suggestion
      description: The format of the built container, oci or docker
```

File: buildah/README.md (hunk 1 above)
Comment (joachimprinzbach) [reply buildah/README.md]: Done, thx

File: buildah/buildah.yaml (hunk 2 above)
Comment (joachimprinzbach) [reply buildah/buildah.yaml]: Done, thx

**Pull request merged**